vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -1257,23 +1337,26 @@ def fused_experts_impl(hidden_states: torch.Tensor,
         curr_topk_ids = topk_ids[begin_chunk_idx:end_chunk_idx]
         curr_topk_weights = topk_weights[begin_chunk_idx:end_chunk_idx]
 
-        a1q_scale: Optional[torch.Tensor] = None
-
-        if use_fp8_w8a8:
-            qcurr_hidden_states, a1q_scale = _fp8_quantize(
-                curr_hidden_states, a1_scale, block_shape)
-        else:
-            qcurr_hidden_states = curr_hidden_states
-            a1q_scale = a1_scale
+        curr_hidden_states, a1_scale = moe_kernel_prepare_input(
```

Comment:
I think you might need to use different variables here since `a1_scale` might get reused (if the loop has multiple iterations).